code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,282 @@
+#define FULL_CRATE_LETTER_ODDS 70
+
+/// Mail is tamper-evident and unresealable, postmarked by CentCom for an individual recepient.
+/obj/item/mail
+	name = "mail"
+	gender = NEUTER
+	desc = "An officially postmarked, tamper-evident parcel regulated by CentCom and made of high-quality materials."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "mail_small"
+	inhand_icon_state = "paper"
+	worn_icon_state = "paper"
+	item_flags = NOBLUDGEON
+	w_class = WEIGHT_CLASS_SMALL
+	drop_sound = 'sound/items/handling/paper_drop.ogg'
+	pickup_sound =  'sound/items/handling/paper_pickup.ogg'
+	mouse_drag_pointer = MOUSE_ACTIVE_POINTER
+	/// Destination tagging for the mail sorter.
+	var/sort_tag = 0
+	/// Who this mail is for and who can open it.
+	var/datum/weakref/recipient
+	/// How many goodies this mail contains.
+	var/goodie_count = 1
+	/// Goodies which can be given to anyone. The base weight for cash is 56. For there to be a 50/50 chance of getting a department item, they need 56 weight as well.
+	var/list/generic_goodies = list(
+		/obj/item/stack/spacecash/c50 = 10,
+		/obj/item/stack/spacecash/c100 = 25,
+		/obj/item/stack/spacecash/c200 = 15,
+		/obj/item/stack/spacecash/c500 = 5,
+		/obj/item/stack/spacecash/c1000 = 1,
+	)
+	// Overlays (pure fluff)
+	/// Does the letter have the postmark overlay?
+	var/postmarked = TRUE
+	/// Does the letter have a stamp overlay?
+	var/stamped = TRUE
+	/// List of all stamp overlays on the letter.
+	var/list/stamps = list()
+	/// Maximum number of stamps on the letter.
+	var/stamp_max = 1
+	/// Physical offset of stamps on the object. X direction.
+	var/stamp_offset_x = 0
+	/// Physical offset of stamps on the object. Y direction.
+	var/stamp_offset_y = 2
+
+	///mail will have the color of the department the recipient is in.
+	var/static/list/department_colors
+
+/obj/item/mail/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	goodie_count = 2
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_MOVABLE_DISPOSING, .proc/disposal_handling)
+	AddElement(/datum/element/item_scaling, 0.5, 1)
+	if(isnull(department_colors))
+		department_colors = list(
+			ACCOUNT_CIV = COLOR_WHITE,
+			ACCOUNT_ENG = COLOR_PALE_ORANGE,
+			ACCOUNT_SCI = COLOR_PALE_PURPLE_GRAY,
+			ACCOUNT_MED = COLOR_PALE_BLUE_GRAY,
+			ACCOUNT_SRV = COLOR_PALE_GREEN_GRAY,
+			ACCOUNT_CAR = COLOR_BEIGE,
+			ACCOUNT_SEC = COLOR_PALE_RED_GRAY,
+		)
+
+	// Icons
+	// Add some random stamps.
+	if(stamped == TRUE)
+		var/N = rand(1, stamp_max)
+		for(var/i = 1, i <= N, i++)
+			var/X = rand(2, 6)
+			stamps += list("stamp_[X]")
```

Comment:
```suggestion
			stamps += list("stamp_[rand(2, 6)]")
```

Reply:
Though these two should probably be defines, if only so that it's easier to change it if more stamps are added.

Plus, what does the `2` mean?

Reply:
Seems to be a protected sprite here I suppose, as stramp_1 is mostly blank looking but may be there for debug reasons.